qiskit/providers/aer/pulse_new/models/operator_models.py

```diff
@@ -0,0 +1,409 @@
+            in_frame_basis (bool): whether to return in the basis in which
+                                        the frame operator is diagonal or not
+
+        Returns:
+            array
+        """
+        # get operators in diagonal frame with signal coefficients applied
+        op_list = vector_apply_diag_frame(t,
+                                          self._operators_in_frame_basis,
+                                          coefficients,
+                                          self._S,
+                                          self._M_cutoff)
+        # generator in diagonal frame_basis
+        op_in_frame_basis = np.sum(op_list, axis=0) - np.diag(self.frame_diag)
+
+        if in_frame_basis:
+            return op_in_frame_basis
+        else:
+            return self.frame_basis @ op_in_frame_basis @ self.frame_basis_adjoint
+
+    def state_into_frame(self, t, y, y_in_frame_basis=False,
+                                     return_in_frame_basis=False):
+        """Take a state into the frame, i.e. return exp(-Ft) @ y.
+
+        Args:
+            t (float): time
+            y (array): state (array of appropriate size)
+            y_in_frame_basis (bool): whether or not the array y is already in
+                                     the frame basis
+            return_in_frame_basis (bool): whether or not to return the result
+                                          in the frame basis
+        """
+
+        out_in_fb = None
+        if y_in_frame_basis:
+            out_in_fb = np.exp(- t * self.frame_diag) * y
+        else:
+            out_in_fb = (np.exp(- t * self.frame_diag) *
+                               (self.frame_basis_adjoint @ y))
+
+        if return_in_frame_basis:
+            return out_in_fb
+        else:
+            return self.frame_basis @ out_in_fb
+
+    def state_out_of_frame(self, t, y, y_in_frame_basis=False,
+                                       return_in_frame_basis=False):
+        """Bring a state out of the frame, i.e. return exp(Ft) @ y.
+
+        Args:
+            t (float): time
+            y (array): state (array of appropriate size)
+            y_in_frame_basis (bool): whether or not the array y is already in
+                                     the frame basis
+            return_in_frame_basis (bool): whether or not to return the result
+                                          in the frame basis
+        """
+
+        out_in_fb = None
+        if y_in_frame_basis:
+            out_in_fb = np.exp(t * self.frame_diag) * y
+        else:
+            out_in_fb = np.diag(np.exp(t * self.frame_diag)) @ self.frame_basis_adjoint @ y
+
+        if return_in_frame_basis:
+            return out_in_fb
+        else:
+            return self.frame_basis @ out_in_fb
+
+
+def vector_apply_diag_frame(t, mats_in_frame_basis, coeffs, S, M_cutoff=None):
+    """Given a list of matrices specified in the frame_basis for a
+    """
```

Comment:
Incomplete docstring

Reply:
I've eliminated this function and just moved the computations into the `FrameFreqHelper.evaluate` method - it was too weird to explain out of context.